components/coding/MCQuiz.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from "react";
+import { GuessData } from "../../pages/api/guessData";
+import {
+  MultipleChoiceSentence,
+  MultipleChoiceSentenceProp,
+} from "../questionTypes/MultipleChoiceSentence";
+
+export interface MCQuiz {
+  questionData: MultipleChoiceSentenceProp[];
+}
+
+export const MCQuiz: React.FC<MCQuiz> = ({ questionData }) => {
+  const [index, setIndex] = useState(0);
+  const [isShaking, setIsShaking] = useState(false);
+  const [notComplete, setNotComplete] = useState(true);
+
+  const submitGuessRequested = (guessData: GuessData) => {
+    setIsShaking(false);
+    if (guessData.isCorrect) {
+      setIndex(index + 1);
+      if (index == questionData.length - 1) {
+        setNotComplete(false);
+      }
+    } else {
+      setIsShaking(true);
+    }
+  };
+
+  return (
+    <>
+      <div
+        className={`rounded-xl shadow-xl mt-4 p-4 transform transition-all bg-blue-100 ${
+          isShaking ? "animate-shake" : ""
+        } `}
+        onAnimationEnd={() => {
+          setIsShaking(false);
+        }}
+      >
+        <div>
+          {notComplete ? (
+            questionData[index].image ? (
```

Comment:
I feel like this ternary isn't needed though? Can you just pass down `questionData[index].image` no matter what. And sometimes it'll just be undefined and in that case the image just won't render?